tests/ops/functions/test_assert_valid.py

```diff
@@ -352,6 +356,12 @@ def create_op_instance(c, str_wires=False):
 
 
 @pytest.mark.jax
+@pytest.fixture(scope="function", autouse=True)
+def capture_warnings(recwarn):
+    """Capture warnings."""
+    yield
+
+
```

Comment:
This makes me rather nervous.  I woudl prefer to expliclty catch expected errors with a `with pytest.warns` context manager.

Reply:
This test needs to capture warnings from other templates. If I used assert as in other examples, it absorbed them and did not raise the error

Reply:
Other templates are raising errors about `QubitStateVector`? Can we remove the use of `QubitStateVector` in those other templates?